color.d

```diff
@@ -1777,14 +1779,53 @@ struct Point {
 	Point opBinary(string op)(int rhs) @nogc {
 		return Point(mixin("x" ~ op ~ "rhs"), mixin("y" ~ op ~ "rhs"));
 	}
+
+	Size opCast(T = Size)() inout @nogc {
+		return Size(x, y);
+	}
 }
 
 ///
 struct Size {
 	int width; ///
 	int height; ///
 
-	int area() pure nothrow @safe const @nogc { return width * height; }
+	pure nothrow @safe:
+
+	/++
+		Rectangular surface area
+
+		Calculates the surface area of a rectangle with dimensions equivalent to the width and height of the size.
+	 +/
+	int area() const @nogc { return width * height; }
+
+	///
+	Point opCast(T = Point)() inout @nogc {
+		return Point(width, height);
+	}
+}
+
+/++
+	Calculates the linear offset of a point
+	from the start (0/0) of a rectangle.
+
+	This assumes that (0/0) is equivalent to offset `0`.
+	Each step on the x-coordinate advances the resulting offset by `1`.
+	Each step on the y-coordinate advances the resulting offset by `width`.
+
+	This function is only defined for the 1st quadrant,
+	i.e. both coordinates (x and y) of `pos` are positive.
+
+	Returns:
+		y × width + x
+ +/
+int linearOffset(const Point pos, const int width) @safe pure nothrow @nogc {
```

Comment:
So, because GitHub doesn’t show this nicely, *here* is the new explanation.

Reply:
yeah should be o k i might do later edits but good enough for now

Reply:
feel free to adjust or improve :)